README.md

```diff
@@ -43,4 +44,63 @@
 
 - Development of this project is automated using [GitHub Actions](https://docs.github.com/en/actions).
 - The `main` branch is protected and pushes to it can only be done through [pull-requests](https://docs.github.com/en/pull-requests) that need to pass the automated branch verification and reviewed by a team member.
-- The verification job in the [CI](https://www.geeksforgeeks.org/what-is-ci-cd/) pipeline runs [linters](https://www.testim.io/blog/what-is-a-linter-heres-a-definition-and-quick-start-guide/), tests and possible builds on both frontend and backend.
+- The verification job in the [CI](https://www.geeksforgeeks.org/what-is-ci-cd/) pipeline runs [linters](https://www.testim.io/blog/what-is-a-linter-heres-a-definition-and-quick-start-guide/), tests and possible builds on both frontend and backend. It also starts the server and runs [End-To-End (E2E) tests](https://katalon.com/resources-center/blog/end-to-end-e2e-testing) on it.
+- The CI pipeline runs also when the changes are merged to `main` branch, and in addition to the `verification` job, it has two jobs that are **only** run when `main` is merged: `tag_release` and `publish_docker_image`. After the `main` branch is verified, the commit is tagged with a [sematic version](https://www.geeksforgeeks.org/introduction-semantic-versioning/). After that, and a [docker image](https://www.techtarget.com/searchitoperations/definition/Docker-image) is created and published to [ghcr.io](https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-container-registry) with the same sematic version tag that the commit was tagged with. It can then be found in the repository packages.
```

Comment:
``` suggestion
- The CI pipeline runs also when the changes are merged to `main` branch, and in addition to the `verification` job, it has two jobs that are **only** run when `main` is merged: `tag_release` and `publish_docker_image`. After the `main` branch is verified, the commit is tagged with a [sematic version](https://www.geeksforgeeks.org/introduction-semantic-versioning/). After that, a [docker image](https://www.techtarget.com/searchitoperations/definition/Docker-image) is created and published to [ghcr.io](https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-container-registry) with the same sematic version tag that the commit was tagged with. It can then be found in the repository packages.
```

Reply:
remove the and